packages/common/src/ContractInteractor.ts

```diff
@@ -308,26 +311,62 @@ export class ContractInteractor {
     }
   }
 
+  async _validateERC165Interfaces (): Promise<void> {
+    if (this.initialisedERC165InstancesCount === 0) {
+      throw new Error('ERC-165 interface check failed. Not a single contract instance initialized')
+    }
+    this.logger.debug(`ERC-165 interface IDs: ${JSON.stringify(erc165Interfaces)}`)
+    const fw = await this._trySupportsInterface(this.forwarderInstance, erc165Interfaces.forwarder)
+    const pm = await this._trySupportsInterface(this.paymasterInstance, erc165Interfaces.paymaster)
+    const pn = await this._trySupportsInterface(this.penalizerInstance, erc165Interfaces.penalizer)
```

Comment:
1. No, the RelayHub does not check the interfaces when these are set. There are four reasons we don't want the checks in solidity constructor: 
 A. We may one day want to use a counterfactual deployment address, and there is no way to do so if you have a call to the method of a contract.
 B. Our tests very frequently pass `address(0)` to constructors and adding this check will require adding the real `StakeManger`, `RelayRegistrar` and `Penalizer` deployment to all tests.
 C. We do not actually have an implemented `IBathchForwarder`. This project is stalled and we may want to remove this parameter. But for now we will be left with a single contract not checked in the constructor anyways.
 D. It is our deployment script that does the deploying, so we can make a check off-chain on deployment anyway. 
 
 
 2. Contracts that are not initialized are not checked. In case we actually implement a post-failure diagnostics method, this checks may be useful.

3. The client may use this method as is. I believe it does not add too much overhead for the client.

Reply:
talked about it with shahaf:
should separate client-verification from sever verification.
the fact CI is used by both client and server, and each initialize different members does NOT mean we need to rely on it.

- server need to verify relayHub members (hub, registrar, penalizer, sm - since the constructor of RH doesn't check for them)
- client should verify the paymaster - since the PM verifies the hub and forwarder. this is sanity protection to detect misconfigured  components between different GSN versions, not an "attack" test of any kind.

(note that "friendly logging" issue is there to verify **developer misconfiguration**, like different forwarder for pm and recipient, etc)

Reply:
I still think that re-using the same method on client and on server, where it does completely different things, is a bad idea: 
relayer never checks paymaster, forwarder (only penalizer, stakemanager, registrar, relayhub)
client only check paymaster, forwarder, 

its true we have a shared object (contractInteractor), but it should not be abused